### ISSUE TITLE
Seamless JetBrains apps do not gain focus v2

### ISSUE THREAD

**Issue by stdedos** [Xpra-org/xpra#3204]
_From https://github.com/Xpra-org/xpra/issues/2852#issuecomment-879688781_:

After having solved https://github.com/Xpra-org/xpra/issues/2852, the issue re-emerged.

However, it doesn't seem to be the same issue; `-d focus` gives:
```
2021-07-14 10:53:27,470 process_focus: wid=15
2021-07-14 10:53:27,470 focus wid=15 has_focus=7
2021-07-14 10:53:27,470 focus: giving focus to WindowModel(0x1410505)
2021-07-14 10:53:27,470 Giving focus to 0x1410505, input_field=0, FORCE_XSETINPUTFOCUS=True, protocols=['WM_DELETE_WINDOW', 'WM_TAKE_FOCUS']
2021-07-14 10:53:27,470 ... using XSetInputFocus
2021-07-14 10:53:27,470 ... using WM_TAKE_FOCUS
2021-07-14 10:53:27,470 sending WM_TAKE_FOCUS: 0x1410505, X11 timestamp=4067517476
2021-07-14 10:53:27,471 focus: will set modified mask to (b'mod2',) using <bound method InputMixin.make_keymask_match of ClientConnectionMuxer(2 : Protocol(unix-domain socket:/run/user/1000/xpra/u-h-2))>
2021-07-14 10:53:27,647 focus wid=15 has_focus=15
2021-07-14 10:53:27,648 focus wid=15 has_focus=15
2021-07-14 10:53:28,153 focus wid=15 has_focus=15
2021-07-14 10:53:28,289 focus wid=15 has_focus=15
2021-07-14 10:53:28,324 focus wid=15 has_focus=15
2021-07-14 10:53:28,345 focus wid=15 has_focus=15
2021-07-14 10:53:28,425 focus wid=15 has_focus=15
2021-07-14 10:53:28,443 focus wid=15 has_focus=15
2021-07-14 10:53:28,462 focus wid=15 has_focus=15
2021-07-14 10:53:28,570 focus wid=15 has_focus=15
2021-07-14 10:53:28,590 focus wid=15 has_focus=15
2021-07-14 10:53:28,590 focus wid=15 has_focus=15
2021-07-14 10:53:28,670 focus wid=15 has_focus=15
2021-07-14 10:53:28,689 focus wid=15 has_focus=15
2021-07-14 10:53:28,706 focus wid=15 has_focus=15
2021-07-14 10:53:28,788 focus wid=15 has_focus=15
2021-07-14 10:53:28,788 focus wid=15 has_focus=15
2021-07-14 10:53:28,804 focus wid=15 has_focus=15
2021-07-14 10:53:28,872 focus wid=15 has_focus=15
2021-07-14 10:53:28,891 focus wid=15 has_focus=15
2021-07-14 10:53:28,909 focus wid=15 has_focus=15
2021-07-14 10:53:28,975 focus wid=15 has_focus=15
2021-07-14 10:53:28,992 focus wid=15 has_focus=15
2021-07-14 10:53:28,993 focus wid=15 has_focus=15
2021-07-14 10:53:29,067 focus wid=15 has_focus=15
2021-07-14 10:53:29,085 focus wid=15 has_focus=15
2021-07-14 10:53:29,103 focus wid=15 has_focus=15
2021-07-14 10:53:29,146 focus wid=15 has_focus=15
2021-07-14 10:53:29,180 focus wid=15 has_focus=15
2021-07-14 10:53:29,253 focus wid=15 has_focus=15
2021-07-14 10:53:30,682 process_focus: wid=0
2021-07-14 10:53:30,682 focus wid=0 has_focus=15
2021-07-14 10:53:30,682 reset_focus() 15 / WindowModel(0x1410505) had focus (toplevel=WorldWindow(0x2c00009))
2021-07-14 10:53:30,683 reset_x_focus: widget with focus: None
2021-07-14 10:53:30,683 Take Focus -> world window
2021-07-14 10:53:30,683 sending WM_TAKE_FOCUS: 0x2c00009, X11 timestamp=4067520688
2021-07-14 10:53:30,774 process_focus: wid=1
2021-07-14 10:53:30,774 focus wid=1 has_focus=0
```

and ~~I guess you don't see something~~ there's nothing out of the ordinary here ~~, right?~~
(and I also see `WM_TAKE_FOCUS` being sent, so I guess nothing).

Logs with `-d focus -d keyboard` are here: [redact-xpra-3204-display-_2-20210714-105155.log](https://github.com/Xpra-org/xpra/files/6821322/redact-xpra-3204-display-_2-20210714-105155.log)

Only the last "asd" text was typed into the window, none of the other mumbo-jumbo I typed

**Comment by totaam** [comment]
Wow, that's a big log file. Do you happen to have `xpra top` connected at the same time?
There are tons of `keyboard info` requests.

There's a focus event just before `dasd`.

Beyond that, not much for me to go at.
I can't really fix this if I can't reproduce it. Please make sure you're running 43c17cada941c28ce32b5ead4bb240eaf6463720 or later.

**Comment by stdedos** [comment]
> Wow, that's a big log file. Do you happen to have `xpra top` connected at the same time?
> There are tons of `keyboard info` requests.

Yes, as always since v4 😀 
The raw quality/speed/latency metrics (even though they are consistently green) are indispensable

> There's a focus event just before `dasd`.

Technically, all I saw typed was `asd`. Keystrokes before that never appeared; as if they were typed in a void.

> I can't really fix this if I can't reproduce it. Please make sure you're running [43c17ca](https://github.com/Xpra-org/xpra/commit/43c17cada941c28ce32b5ead4bb240eaf6463720) or later.

Full session log (hopefully not as verbose): [redact-display-_2-20210716-100108.log](https://github.com/Xpra-org/xpra/files/6828588/redact-display-_2-20210716-100108.log)

_And I am downgrading (https://github.com/Xpra-org/xpra/issues/3206)_

**Comment by stdedos** [comment]
In r29499 (27a2d4a) it seems to have disappeared 🤷 (fixed by https://github.com/Xpra-org/xpra/issues/3202?)

**Comment by totaam** [comment]
> fixed by #3202?

That's possible I guess. I'm closing this ticket, feel free to re-open it.
